Source/WebPImage.swift

```diff
@@ -16,11 +16,16 @@ public class WebPImageDecoder: Nuke.ImageDecoding {
     public init() {
     }
 
-    public func decode(data: Data, isFinal: Bool) -> Image? {
+    public func decode(_ data: Data) -> ImageContainer? {
         guard data.isWebPFormat else { return nil }
-        guard !isFinal else { return _decode(data) }
+        guard let image = _decode(data) else { return nil }
+        return ImageContainer(image: image)
+    }
 
-        return decoder.incrementallyDecode(data, isFinal: isFinal)
+    public func decodePartiallyDownloadedData(_ data: Data) -> ImageContainer? {
+        guard data.isWebPFormat else { return nil }
+        guard let image = decoder.incrementallyDecode(data) else { return nil }
```

Comment:
(A suggestion, not in the scope of PR) There might be an optimization potential here. If you take progressive JPEG, it has indicators where one scan ends and another begins. By scanning the JPEG, you can avoid producing the same scans multiple times. I'm doing this in [ProgressiveJPEGScanner](https://github.com/kean/Nuke/blob/master/Sources/ImageDecoding.swift#L148) which can potentially be generalized.

I'm not sure how much impact it really has, I implemented it a while ago and don't remember the test results, I think I should revisit it.

Reply:
@kean Thank you!
In this PR, we will follow Nuke 9 and release it, and I will improve it with another PR referring to ProgressiveJPEGScanner!
Please review at that time 😆